src/main/kotlin/by/architecture_map/belarus/service/impl/ArchitecturalStyleServiceImpl.kt

```diff
@@ -0,0 +1,69 @@
+package by.architecture_map.belarus.service.impl
+
+import by.architecture_map.belarus.entity.ArchitecturalStyle
+import by.architecture_map.belarus.exception.NotFoundException
+import by.architecture_map.belarus.repository.ArchitecturalStyleRepository
+import by.architecture_map.belarus.service.ArchitecturalStyleService
+import by.architecture_map.belarus.service.ImageService
+import org.springframework.stereotype.Service
+
+@Service
+class ArchitecturalStyleServiceImpl(
+    private val architecturalStyleRepository: ArchitecturalStyleRepository,
+    private val imageService: ImageService
+) : ArchitecturalStyleService {
+
+    override fun create(architecturalStyle: ArchitecturalStyle) = architecturalStyleRepository.save(architecturalStyle)
+
+    override fun findAll() = architecturalStyleRepository.findAll()
+
+    override fun find(id: Int): ArchitecturalStyle = architecturalStyleRepository.findById(id)
+        .orElseThrow { throw NotFoundException("Architectural style not found with id: $id") }
+
+    override fun update(id: Int, updatedArchitecturalStyle: ArchitecturalStyle): ArchitecturalStyle =
+        architecturalStyleRepository.findById(id)
```

Comment:
please try not to create code duplicates and always reuse existing code. in that case you have three methods that implements the same search by id logic and two update methods that are almost the same.

here is an example how to refactor everything:
implement a common private update method:
```
    private fun applyUpdates(id: Int, update: ArchitecturalStyle.() -> Unit) = find(id)
        .apply(update)
        .let { architecturalStyleRepository.save(it) }
```

and the method itself:
```
    override fun update(id: Int, updatedArchitecturalStyle: ArchitecturalStyle): ArchitecturalStyle =
        applyUpdates(id) {
            name = updatedArchitecturalStyle.name
            attributes = updatedArchitecturalStyle.attributes
            description = updatedArchitecturalStyle.description
            shortDescription = updatedArchitecturalStyle.shortDescription
            demonstrativeImage = updatedArchitecturalStyle.demonstrativeImage
            yearsActive = updatedArchitecturalStyle.yearsActive
        }
```